twisted/test/test_text.py

```diff
@@ -181,66 +181,66 @@ def test_multiWord(self):
 
 class StrFileTests(unittest.TestCase):
     def setUp(self):
-        self.io = StringIO("this is a test string")
+        self.stream = StringIO(u"this is a test string")
```

Comment:
The tests below seem to indicate that `strFile` searches a file for a string of whatever type is native to the current version of Python.  But as is clear here, `io.StringIO` operates on `unicode` objects in Python 2.  That means that all the test data below gets implicitly decoded using the default encoding (specifically, it's happening on [these lines](https://github.com/twisted/twisted/blob/trunk/twisted/python/text.py#L203-L206)).

These tests should use the `io` object whose `.read()` method returns native strings.  In Python 2 that will be a `BytesIO` object, while on Python 3 it'll be a `StringIO`:

```python
test_text = "this is a test string"
if _PY3:
    self.stream = io.StringIO(test_text)
else:
    self.stream = io.BytesIO(test_text)
````

What do you think about a follow-up ticket that changes `strFile`'s docstring to more explicitly describe the string type it wants?